### ISSUE TITLE
FullTextIndexAttribute support

### ISSUE THREAD

**Issue by borisdj** [isukces/EfCore.Shaman#11]
FullText Search is another sometimes often used feature so it would be useful to have `[FullTextIndexAttribute]` for it as well.
At the moment I am setting it manually in migration with raw _Sql_ (encapsulated in function):
migrationBuilder.Sql(FullTextSql.CreateIndex(nameof(Customer.FirstName), nameof(Customer)), true);
protected override void Up(MigrationBuilder migrationBuilder)
```
{
    migrationBuilder.Sql(FullTextSql.DropIndex(nameof(Customer.FirstName)), true);
}
protected override void Down(MigrationBuilder migrationBuilder)
{
    migrationBuilder.Sql(FullTextSql.DropIndex(nameof(Customer.FirstName)), true);
 }
```

And these function are defined in following class:
(Remark: FullTextSql.CreateIndex requires that CATALOG exist which requires STOPLIST, and they are created in the first migration. That requirement(previous manual creation of StopList nad Catalog) can stay the same, I am not sure how this could be done automatically, since they only need to be created once and all FTIndexes can use them)

```
public static class FullTextSql
{
    public static string CreateIndex(string columnName, string tableName, string shema = "dbo")
    {
        return $"CREATE FULLTEXT INDEX ON {shema}.{tableName}({columnName}) KEY INDEX PK_{tableName} WITH STOPLIST = EmptyStopList;";
    }
    public static string DropIndex(string tableName, string shema = "dbo")
    {
        return $"DROP FULLTEXT INDEX ON {shema}.{tableName};";
    }
    public static string CreateCatalog() { return "CREATE FULLTEXT CATALOG FTC AS DEFAULT;"; }
    public static string CreateStoplist() { return "CREATE FULLTEXT STOPLIST EmptyStopList;"; }

    public static string DropCatalog() { return "DROP FULLTEXT CATALOG FTC;"; }
    public static string DropStoplist() { return "DROP FULLTEXT STOPLIST EmptyStopList;"; }
}
```

**Comment by isukces** [comment]
I've just started to implement this feature. 
First I have each time verify if catalog exists on each migration, because I'm unable to obtain information if catalog has been introduced in last or previous migration.
It's not best approach but have no idea how to solve this.

`if not exists (SELECT * FROM sys.fulltext_catalogs where name='my catalog name') CREATE FULLTEXT CATALOG [my catalog name];`

Please check last commit and leave comment.

**Comment by borisdj** [comment]
I can't make FTS attribute work.
When put just `[FullTextIndex]` with no parameters it gives error:
_Unable to find FullTextCatalogName for table Movies_
And if I use `[FullTextIndex(FullTextCatalogName = "FTC")]` I can create migration but Up and Down methods are empty.

**Comment by isukces** [comment]
The problem is that from EntityFramework's point of view there is nothing to migrate. I have no idea at this moment how to fix this.

**Comment by borisdj** [comment]
So currently this attribute is informative only?
Meaning that after manually creating FTS index we put [FullTextIndex] on column as marker.

**Comment by isukces** [comment]
No. It willl affect each next migration, by adding some sql statements. 
Shaman can only fix migrations made by EF. It cannot force EF to create migration if EF doesn't want.

**Comment by borisdj** [comment]
Even when I make another change because of which migration is created, that migration still does not contains anything regarding FullText (FTIndex still being ignored), and if I run update-database there is no index.
Should something else be configured for this ?

**Comment by borisdj** [comment]
Is this feature not implemented yet?
I can see SqlServerFullTextIndexService in code here on github, but after adding nugget in project there is no reference to it.
